collector_test.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package datadog
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+func TestAddViewData(t *testing.T) {
+	exporter := NewExporter(Options{Namespace: "hello", Tags: []string{"test:optionalTag"}})
+
+	view.RegisterExporter(exporter)
+	reportPeriod := time.Millisecond
+	view.SetReportingPeriod(reportPeriod)
+
+	expected := &view.Data{
+		View: newView(view.Count()),
+	}
+	exporter.collector.addViewData(expected)
+	actual := exporter.collector.viewData["hello.fooCount"]
+
+	if actual != expected {
+		t.Errorf("Expected: %v, Got: %v", expected, actual)
+	}
+}
+func TestNilAggregation(t *testing.T) {
+	exporter := NewExporter(Options{})
+	view.RegisterExporter(exporter)
+	reportPeriod := time.Millisecond
+	view.SetReportingPeriod(reportPeriod)
+	noneAgg := &view.Aggregation{
+		Type:    view.AggTypeNone,
+		Buckets: []float64{1},
+	}
+	vd := &view.Data{
```

Comment:
`vd` is not used anywhere, we could just have:
```go
view := customNewView("fooNone", noneAgg, testTags, measureCount)
```
And then pass that to `submitMetric` further down.